.circleci/config.yml

```diff
@@ -557,6 +553,23 @@ jobs:
       ENVOY_VERSIONS: "1.11.1"
     steps: *ENVOY_INTEGRATION_TEST_STEPS
 
+  # run tests on vault ca provider integration tests
+  vault-ca-provider:
+    docker:
+      - image: *GOLANG_IMAGE
+    environment:
+      <<: *ENVIRONMENT
+    steps:
+      # Install vault
+      - run: |
+          wget -q -O /tmp/vault.zip https://releases.hashicorp.com/vault/${VAULT_BINARY_VERSION}/vault_${VAULT_BINARY_VERSION}_linux_amd64.zip
+          sudo unzip -d /usr/local/bin /tmp/vault.zip
+          rm -rf /tmp/vault*
+      # Gather deps to run go tests
+      - checkout
```

Comment:
I think you should copy a snippet from the go-tests target to bootstrap the go mod cache rather than starting from zero modules:
```
 - restore_cache: # restore cache from dev-build job
          keys:
            - consul-modcache-v1-{{ checksum "go.mod" }}
```

Reply:
Added, thanks!!